tools/coredump/modulestore/store.go

```diff
@@ -197,16 +200,26 @@ func (store *Store) UploadModule(id ID) error {
 		return fmt.Errorf("failed to open local file: %w", err)
 	}
 
+	fileData, err := io.ReadAll(file)
```

Comment:
We don't know how large a coredump file in the future can be, how much RAM a user has available or if the coredump already is in a RAM disk.

I'd suggest to hash from disk and also let PutObject stream from disk.
The potential (but extremely unlikely) TOCTOU (data changes between reads) is caught on the server side with the checksum provided.

Reply:
The Oracle backed S3 storage did not accept streams. That's why I made this change alongside the newly required checksum.

Reply:
This sounds weird. How did you test this? The `Body` seems to be working on a reader interface so this really should not matter.

Wild guess: did you use same reader for first hashing and *not* seeking back to beginning so that the http send starts from right position?